costmap_2d/include/costmap_2d/footprint.h

```diff
@@ -120,8 +120,10 @@ bool makeFootprintFromString(const std::string& footprint_string, std::vector<ge
 /**
  * @brief Read the ros-params "footprint" and/or "robot_radius" from
  * the given NodeHandle using searchParam() to go up the tree.
+ * @return The footprint as a list of points and whether we used the robot_radius.
+ * I that's the case, the footprint will be a circle of "robot_radius".
  */
-std::vector<geometry_msgs::Point> makeFootprintFromParams(ros::NodeHandle& nh);
+std::pair<std::vector<geometry_msgs::Point>, bool> makeFootprintFromParams(ros::NodeHandle& nh);
```

Comment:
Q: do we need to make this change? 
we can retrieve the extra info (pair.second) by calling `getUseRadius`, right?

Reply:
yes, but I thought it was more natural this way

Reply:
`std::optional` maybe fits better here?

Reply:
mmm...  I already changed on the client side to use this,,, will keep it as it is